### PR TITLE
Update deps to work with current node

### PR DESCRIPTION
**Pull request by timoxley**
Updates deps so platform can build on node v12/v14.
This updates most deps minor versions, held back flow & some eslint deps as those updates required too much change that wasn't auto-fixable.

### CODE REVIEW

**Review by juhah — APPROVED** [review]
Looks good, tried building with both node 12 & 14, didn't notice anything out of the ordinary. There is one warning coming up when you create a canvas or enter the canvas editor that doesn't come up with the current version:

Screenshot 2021-02-03 at 9 39 52

It's just a warning, though.

**Comment by juhah** [comment]
Also fyi, there is this warning:

Screenshot 2021-02-04 at 13 47 53

**Comment by juhah** [comment]
VScode shows this error:
Screenshot 2021-02-04 at 14 16 54

**Comment by timoxley** [comment]
> VScode shows this error
@juhah That is odd, are you sure VScode is running the eslint version from `node_modules` instead of a globally installed one? To check try `npm uninstall -g eslint`

**Comment by timoxley** [comment]
> Also fyi, there is this warning:

Fixed.

**Comment by juhah** [comment]
>  That is odd, are you sure VScode is running eslint from inside node_modules instead of a globally installed one? To check try npm uninstall -g eslint

@timoxley Maybe it was just a one-off thing, now it works again 🤷

**Review by juhah — APPROVED** [review]
Tests pass, no more warnings or errors 🎉🎉🎉

**Review by tumppi — APPROVED** [review]
Nice! Tested with Node 14 and works perfectly also under WSL on Windows 👍

File: app/src/marketplace/components/GoogleAnalyticsTracker/index.jsx [hunk 1]
```diff
@@ -10,25 +10,24 @@ type Props = {
...
-        if (newProps.location && (!this.props.location || newProps.location.pathname !== this.props.location.pathname)) {
-            this.logPageview(newProps.location.pathname)
-        }
-    }
+    const { pathname = '' } = location
 
-    logPageview = (page) => {
-        ReactGA.pageview(page)
-    }
+    // run once
```
Comment (mondoreale) [inline app/src/marketplace/components/GoogleAnalyticsTracker/index.jsx:20]: 👍

File: app/src/marketplace/utils/smartContract.js [hunk 2]
```diff
@@ -83,16 +83,17 @@ export const send = (method: Sendable, options?: {
 }): SmartContractTransaction => {
     const web3 = getWeb3()
     const emitter = new EventEmitter()
+    // NOTE: looks like there's double handling of errors happening here
+    // i.e. .catch + on('error')
```
Comment (mondoreale) [inline app/src/marketplace/utils/smartContract.js:87]: I wonder if `.on('error')` is "operational" at the point of that promise rejection (race condition). It's worth having a comment here though. Let it help the future us.

File: app/src/shared/components/Calendar/index.jsx [hunk 3]
```diff
@@ -3,22 +3,36 @@
...
-import Arrow from '../CalendarArrowIcon'
 import dateFormatter from '$utils/dateFormatter'
+import Arrow from '../CalendarArrowIcon'
 import styles from './calendar.pcss'
 
 type Props = CalendarProps & {
     onChange?: (Date) => void,
 }
 
+// wrap format functions as react-calendar now passes locale as first param
```
Comment (mondoreale) [inline app/src/shared/components/Calendar/index.jsx:14]: 👌

File: app/travis_scripts/smoke-test.sh [hunk 4]
```diff
@@ -1,12 +1,13 @@
...
+#!/bin/bash
 
 set -e
 
 ## Script for preparing smoke test
 sudo ifconfig docker0 10.200.10.1/24
 ## Get Streamr Docker dev
-mkdir "$TRAVIS_BUILD_DIR/streamr-docker-dev"
-git clone https://github.com/streamr-dev/streamr-docker-dev.git "$TRAVIS_BUILD_DIR/streamr-docker-dev"
+if [ ! -d "$TRAVIS_BUILD_DIR/streamr-docker-dev" ]; then
```
Comment (mondoreale) [inline app/travis_scripts/smoke-test.sh:8]: 👌

**Review by mondoreale — APPROVED** [review]
Hefty PR from a distance, but not so hefty up close. 🐕

Looks good to me, sir. Compiles without a warning. Tried it in the browser, too.

File: app/src/auth/utils/userAuthenticated.js [hunk 5]
```diff
@@ -14,22 +14,12 @@ type Props = {
...
 }
 
-class FailureComponent extends Component<Props> {
-    componentWillMount() {
-        this.redirect()
-    }
+function FailureComponent(props: Props) {
+    useEffect(() => {
+        props.redirect(props, props.redirectPath)
+    }, [props])
```
[the remark this answers is not in the document]
Comment (timoxley) [reply app/src/auth/utils/userAuthenticated.js:20]: Correct. 
I guess the correct way would be to capture the props as a ref then do this but meh.

File: app/src/marketplace/components/GoogleAnalyticsTracker/index.jsx [hunk 6]
```diff
@@ -10,25 +10,24 @@ type Props = {
     location: Location,
 }
 
-class GoogleAnalyticsTracker extends Component<Props> {
-    constructor(props: Props) {
-        super(props)
-        // Must call window.ga('create', gaId) instead of ReactGA.initialize(gaId) since we don't want to inject the ga script to DOM again
-        window.ga('create', gaId)
-        this.logPageview(this.props.location.pathname)
+export default withRouter(({ location = {} }: Props) => {
```
[the remark this answers is not in the document]
Comment (timoxley) [reply app/src/marketplace/components/GoogleAnalyticsTracker/index.jsx:13]: Yeah that would be better I guess, was just doing a direct conversion of the class using what was already available.

File: app/src/marketplace/components/GoogleAnalyticsTracker/index.jsx (hunk 1 above)
Comment (timoxley) [reply app/src/marketplace/components/GoogleAnalyticsTracker/index.jsx:20]: 👍

File: app/src/shared/components/Calendar/index.jsx (hunk 3 above)
Comment (timoxley) [reply app/src/shared/components/Calendar/index.jsx:14]: 👍

File: app/travis_scripts/smoke-test.sh (hunk 4 above)
Comment (timoxley) [reply app/travis_scripts/smoke-test.sh:8]: 👏

**Comment by timoxley** [comment]
CI failing but tests pass locally 🤷  nfi

**Pull request merged**